gradio/components.py

```diff
@@ -444,15 +433,46 @@ def __init__(
         self.cleared_value = ""
         self.test_input = default_value
         self.interpret_by_tokens = True
-        IOComponent.__init__(self, label=label, css=css, **kwargs)
+        IOComponent.__init__(
+            self,
+            label=label,
+            show_label=show_label,
+            interactive=interactive,
+            css=css,
+            visible=visible,
+            **kwargs,
+        )
 
-    def get_template_context(self):
+    def get_config(self):
         return {
             "lines": self.lines,
             "max_lines": self.max_lines,
             "placeholder": self.placeholder,
             "default_value": self.default_value,
-            **IOComponent.get_template_context(self),
+            **IOComponent.get_config(self),
+        }
+
+    @staticmethod
+    def update(
+        value: Optional[Any] = None,
+        lines: Optional[int] = None,
+        max_lines: Optional[int] = None,
+        placeholder: Optional[str] = None,
+        label: Optional[str] = None,
+        show_label: Optional[bool] = None,
+        css: Optional[Dict] = None,
+        visible: Optional[bool] = None,
```

Comment:
You don't have `interactive` as a parameter in any of the `update()` methods -- is that intentional?